internal/engine/experiment/quicping/quicping.go

```diff
@@ -0,0 +1,479 @@
+// Package quicping implements the quicping network experiment. This
+// implements, in particular, v0.1.0 of the spec.
+//
+// See https://github.com/ooni/spec/blob/master/nettests/ts-031-quicping.md.
+package quicping
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"time"
+
+	_ "crypto/sha256"
+
+	"github.com/ooni/probe-cli/v3/internal/engine/netx/archival"
+	"github.com/ooni/probe-cli/v3/internal/model"
+	"github.com/ooni/probe-cli/v3/internal/netxlite"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+)
+
+// A ConnectionID in QUIC
+type ConnectionID []byte
```

Comment:
```suggestion
// A ConnectionID in QUIC
type connectionID []byte
```

This type should be private